tests/Unit/EmailTest.php

```
@@ -0,0 +1,170 @@
+	// -------------------------------------------------------------------------
+	// Placeholder substitution
+	// -------------------------------------------------------------------------
+
+	/** Simple placeholder replaced with scalar data value. */
+	public function test_placeholder_replaced(): void {
+		WP_Mock::userFunction( 'apply_filters' )
+			->andReturnArg( 1 );
+
+		$result = swh_parse_template( 'Hello {name}!', array( 'name' => 'Alice' ) );
+		$this->assertSame( 'Hello Alice!', $result );
+	}
```

PHPMD (2.15.0)

[error] The method test_placeholder_replaced is not named in camelCase. (undefined)

(CamelCaseMethodName)